tests/checks/integration/test_powerdns_recursor.py

```diff
@@ -22,60 +24,196 @@ class TestPowerDNSRecursorCheck(AgentCheckTest):
+        'dnssec-result-nta',
+        'dnssec-result-secure',
+        'dnssec-validations',
+        'dont-outqueries',
+        'edns-ping-matches',
+        'edns-ping-mismatches',
+        'ignored-packets',
+        'ipv6-outqueries',
+        'ipv6-questions',
+        'malloc-bytes',
+        'max-mthread-stack',
+        'no-packet-error',
+        'noedns-outqueries',
+        'noerror-answers',
+        'noping-outqueries',
+        'nsset-invalidations',
+        'nsspeeds-entries',
+        'nxdomain-answers',
+        'outgoing-timeouts',
+        'outgoing4-timeouts',
+        'outgoing6-timeouts',
+        'over-capacity-drops',
+        'packetcache-hits',
+        'packetcache-misses',
+        'policy-drops',
+        'policy-result-custom',
+        'policy-result-drop',
+        'policy-result-noaction',
+        'policy-result-nodata',
+        'policy-result-nxdomain',
+        'policy-result-truncate',
+        'qa-latency',
+        'questions',
+        'real-memory-usage',
+        'resource-limits',
+        'security-status',
+        'server-parse-errors',
+        'servfail-answers',
+        'spoof-prevents',
+        'sys-msec',
+        'tcp-client-overflow',
+        'tcp-clients',
+        'tcp-outqueries',
+        'tcp-questions',
+        'throttled-out',
+        'throttled-outqueries',
+        'too-old-drops',
+        'udp-in-errors',
+        'udp-noport-errors',
+        'udp-recvbuf-errors',
+        'udp-sndbuf-errors',
+        'unauthorized-tcp',
+        'unauthorized-udp',
+        'unexpected-packets',
+        'unreachables',
+        'uptime',
+        'user-msec',
     ]
 
+
     METRIC_FORMAT = 'powerdns.recursor.{}'
 
     def __init__(self, *args, **kwargs):
         AgentCheckTest.__init__(self, *args, **kwargs)
-        self.config = {"instances": [{
+        self.config_v3 = {"instances": [{
             "host": "127.0.0.1",
             "port": "8082",
             "api_key": "pdns_api_key"
         }]}
+        self.config_v4 = {"instances": [{
+            "host": "127.0.0.1",
+            "port": "8083",
+            "api_key": "pdns_api_key",
+            "version": 4
+        }]}
 
     # Really a basic check to see if all metrics are there
     def test_check(self):
-        self.run_check_twice(self.config)
+        # Run Version 3
+        self.run_check_twice(self.config_v3)
 
         # Assert metrics
-        for metric in self.GAUGE_METRICS:
+        for metric in self.GAUGE_METRICS_V3:
             self.assertMetric(self.METRIC_FORMAT.format(metric), tags=[])
 
-        for metric in self.RATE_METRICS:
+        for metric in self.RATE_METRICS_V3:
             self.assertMetric(self.METRIC_FORMAT.format(metric), tags=[])
 
         service_check_tags = ['recursor_host:127.0.0.1', 'recursor_port:8082']
         self.assertServiceCheckOK('powerdns.recursor.can_connect', tags=service_check_tags)
 
         self.coverage_report()
 
+
+    def test_check_v4(self):
```

Comment:
I am unsure what you are referring to here. 

When I run version 4 and version 3 locally with the configuration defined in self.config_v3 and self.config_v4 (both versions running on different ports) the tests pass locally. Would you be able to elaborate further on what your expectations are?

```
# nosetests tests/checks/integration/test_powerdns_recursor.py
......
----------------------------------------------------------------------
Ran 6 tests in 3.164s

OK

# curl -H 'X-API-Key: pdns_api_key' -s http://127.0.0.1:8083/api/v1/servers/localhost | jq
{
  "config_url": "/api/v1/servers/localhost/config{/config_setting}",
  "daemon_type": "recursor",
  "id": "localhost",
  "type": "Server",
  "url": "/api/v1/servers/localhost",
  "version": "0.0.g9413017",
  "zones_url": "/api/v1/servers/localhost/zones{/zone}"
}

# curl -H 'X-API-Key: pdns_api_key' -s http://127.0.0.1:8082/servers/localhost | jq
{
  "type": "Server",
  "id": "localhost",
  "url": "/servers/localhost",
  "daemon_type": "recursor",
  "version": "3.7.3",
  "config_url": "/servers/localhost/config{/config_setting}",
  "zones_url": "/servers/localhost/zones{/zone}"
}
```

I am keen to ensure that we are on the same page so there is no confusion.